includes/um-short-functions.php

```diff
@@ -1551,19 +1551,13 @@ function um_can_view_field( $data ) {
 					}
 				}
 				break;
-			case '-3':
+			case '-3': // Only visible to profile owner and specific roles
 				if ( ! is_user_logged_in() ) {
 					$can_view = false;
-				} else {
-					if ( ! um_is_core_page( 'profile' ) ) {
-						if ( empty( $current_user_roles ) || ( ! empty( $data['roles'] ) && count( array_intersect( $current_user_roles, $data['roles'] ) ) <= 0 ) ) {
-							$can_view = false;
-						}
-					} else {
-						if ( ! um_is_user_himself() && ( empty( $current_user_roles ) || ( ! empty( $data['roles'] ) && count( array_intersect( $current_user_roles, $data['roles'] ) ) <= 0 ) ) ) {
-							$can_view = false;
-						}
-					}
+				} elseif ( um_is_core_page( 'user' ) && um_is_user_himself() ) {
+					$can_view = true;
+				} elseif ( empty( $current_user_roles ) || ( ! empty( $data['roles'] ) && count( array_intersect( $current_user_roles, $data['roles'] ) ) <= 0 ) ) {
+					$can_view = false;
```

Comment:
Please check if the user has  necessary roles and set $can_view = true;

or set $can_view = false; in line 1555 and rewrite some conditions

Reply:
I don't understand what is wrong:
- It is impossible to check if the user has necessary roles if the user is logged out.
- There is no sense to check if the user has necessary roles if the user is a profile owner. Owners have to see the field no matter what their role is.
- `$can_view = true;` is the default value,

We have to set $can_view = false; if the user is logged out or if logged in user is not a profile owner and has not necessary roles.

Reply:
Hi @yuriinalivaiko  @nikitasinelnikov  Could you also check this privacy issue with the fields? 
https://wordpress.org/support/topic/front-end-40/